lib/application.js

```diff
@@ -567,3 +571,65 @@ app.listen = function(cb) {
 
   return server;
 };
+
+/* START CONTROLLER CODE*/
+
+// this is similar to app.model
+// at the moment I haven't really used 'options'
+app.controller = function(ctor, options) {
+  var self = this;
+
+  // the SharedController is a wrapper, see below
+  var controller = new SharedController(ctor);
+  this.remotes().addClass(controller.sharedClass);
+  app.controllers().push(controller);
+
+  return controller;
+
+  // SharedController class contains strong-remoting objects to set up the metadata
+  // and register the endpoint, etc
+  function SharedController(ctor, options) {
+    options = options || {};
+    this.name = ctor.name;
+    this.ctor = ctor;
+    // TODO - right now there's a hack on line 289 to skip this
+    // util.inherits(ctor, EventEmitter);
+
+    this.sharedClass = new RemoteObjects.SharedClass(this.name, ctor, options);
+    // the resolveScope method instantiates the controller upon request
+    // (please correct my terminology if it's wrong)
+    this.sharedClass.resolveScope = function(remotingContext) {
+      // self is the app - should it be something more obvious like appInstance?
+      return new ctor(self, remotingContext);
+    };
+
+    // similar to model.remoteMethod
+    this.remoteMethod = function(name, options) {
+      // I didn't dive into why, but this only worked when I set isStatic to true
+      // whether the controller methods were prototype methods or not
+      options.isStatic = true;
+      this.sharedClass.defineMethod(name, options);
+    };
+
+    // have not tested this method yet but it follows model.disableRemoteMethod
+    this.disableRemoteMethod = function(name, isStatic) {
+      this.sharedClass.disableMethod(name, isStatic || false);
+      // this.emit('remoteMethodDisabled', this.sharedClass, name);
```

Comment:
The event `remoteMethodDisabled` is used by loopback-component-explorer to rebuild the list of methods displayed in the UI, see https://github.com/strongloop/loopback/pull/2266/files and https://github.com/strongloop/loopback-component-explorer/pull/157/files.

Perhaps we can emit the event directly on the `app` object?

Alternatively, we can change `SharedController` to inherit from `EventEmitter`, then here we can emit the event on the shared-controller instance and finally `app.controller` can install a listener for this event that will emit app-wide event, similarly to what we do in https://github.com/strongloop/loopback/pull/2266/files.

Thoughts?